OpenBCISample.js

```diff
@@ -0,0 +1,200 @@
+const SCALE_FACTOR_ACCEL = 0.002 / Math.pow(2,4);
+// Scale factor for channelData
+const SCALE_FACTOR_CHANNEL = ADS1299_VREF / ADS1299_GAIN / (Math.pow(2,23) - 1);
+
+var k = require('./OpenBCIConstants');
+
+/*
+ Errors
+ */
+const kErrorInvalidByteLength = "Invalid Packet Byte Length";
+const kErrorInvalidByteStart = "Invalid Start Byte";
+const kErrorInvalidByteStop = "Invalid Stop Byte";
+
+module.exports = {
+    convertPacketToSample: function (dataBuf) {
+        var self = this;
+        if(dataBuf === undefined || dataBuf === null) {
+            reject('data is undefined');
+        }
+        var numberOfBytes = dataBuf.byteLength;
+        var scaleData = true;
+
+        if (dataBuf[0] != BYTE_START) return;
+        if (dataBuf[32] != BYTE_STOP) return;
+        if (numberOfBytes != k.OBCIPacketSize) return;
+
+        var channelData = function () {
+            var out = {};
+            var count = 1;
+            for (var i = 2; i <= numberOfBytes - 10; i += 3) {
+                //console.log('\tDataBuf is is: ' + dataBuf.slice(i, i + 3).toString('hex'));
+                //var newNumber = self.interpret24bitAsInt32(dataBuf.slice(i, i + 3));
+                //out[count] = newNumber * SCALE_FACTOR_CHANNEL;
+                out[count] = scaleData ? self.interpret24bitAsInt32(dataBuf.slice(i, i + 3)) * SCALE_FACTOR_CHANNEL : self.interpret24bitAsInt32(dataBuf.slice(i, i + 3));
+                //console.log("in" + dataBuf.slice(i,i+3));
+                //console.log(out[count]);
+                count++;
+            }
+            return out;
+        };
+
+        var auxData = function () {
+            var out = {};
+            var count = 0;
+            for (var i = numberOfBytes - 7; i < numberOfBytes - 1; i += 2) {
+                out[count] = scaleData ? self.interpret16bitAsInt32(dataBuf.slice(i, i + 2)) * SCALE_FACTOR_ACCEL : self.interpret16bitAsInt32(dataBuf.slice(i, i + 2));
+                count++;
+            }
+            return out;
+        };
+
+        return {
+            startByte: dataBuf[0], //byte
+            sampleNumber: dataBuf[1], //byte
+            channelData: channelData(), // multiple of 3 bytes
+            auxData: auxData(), // 6 bytes
+            stopByte: dataBuf[numberOfBytes - 1] //byte
+        }
+    },
+    debugPrettyPrint: function(sample) {
+        console.log('-- Sample --');
+        console.log('---- Start Byte: ' + sample.startByte);
+        console.log('---- Sample Number: ' + sample.sampleNumber);
+        for(var i = 0; i < 8; i++) {
+            console.log('---- Channel Data ' + i + ': ' + sample.channelData[i]);
+        }
+        for(var j = 0; j < 3; j++) {
+            console.log('---- Aux Data ' + j + ': ' + sample.auxData[j]);
+        }
+        console.log('---- Stop Byte: ' + sample.stopByte);
+
+
+    },
+    scaleFactorAux: SCALE_FACTOR_ACCEL,
+    scaleFactorChannel: SCALE_FACTOR_CHANNEL,
+    sampleMaker: function(length) {
```

Comment:
So this whole section has to do with defining an OpenBCISample, and the properties and methods that go along with it. I have it being tested by OpenBCISample-test.js.